### ISSUE TITLE
Setup i2p tunnels

### ISSUE THREAD

**Issue by nahuhh** [Gedsh/InviZible#211]
Trying to setup i2p tunnels (tunnels.conf) and need to input the location of the key.dat files.
If this is currently possible, where do i put the files?
https://github.com/Gedsh/InviZible/issues/21#issuecomment-601686980

if not, would it be a good idea to store the config files on public storage similar to i2pd app?

scrrenshot attached 
![zzssbai1prnr1r0f.jpg](https://user-images.githubusercontent.com/50635951/229982444-420acd65-1f1c-421e-b52d-1d76c97bec61.jpg)

**Comment by nahuhh** [comment]
Yes. It appears you must put files in `/data/user/0/pan.alexander.tordnscrypt/i2pd_data` (inaccessible)

please add a storage permission to allow access to `/storage/emulated/0/invizible` or `/storage/emulated/0/`

**Comment by Gedsh** [comment]
> would it be a good idea to store the config files on public storage

This is a bad idea as it raises a lot of privacy issues. Why a privacy-focused app needs full storage access. It also requires MANAGE_EXTERNAL_STORAGE permission to work on modern android versions. I have to find a better and privacy oriented solution.

But for now, you can try save a backup. It is a zip archive. Add the necessary files and restore the backup.

**Comment by nahuhh** [comment]
will try and get back to you within 48hrs

**Comment by nahuhh** [comment]
48 hrs was a little long

the backup restore method doesnt work due to
1. backup doesnt include the i2pd folder, and only the i2pd_data folder
2. adding a tor hidden service complains about the folder being too permissive

maybe a workaround for 1. can let invizible import i2p .dat key files (and save them in the inaccessible i2pd folder)
or a backup that includes the excluded folder?

not sure about tor.

**Comment by Gedsh** [comment]
> backup doesnt include the i2pd folder, and only the i2pd_data folder

Can't you use the i2pd_data folder? It should be accessible from i2pd as it is the app's internal folder.

> adding a tor hidden service

Hosting hidden services is not planned.
